spec/queries/credit_notes_query_spec.rb

```diff
@@ -362,6 +362,24 @@
     end
   end
 
+  context "when billing entity ids filter applied" do
+    let(:billing_entity) { create(:billing_entity, organization:) }
+    let(:filters) { {billing_entity_ids: [billing_entity.id]} }
+
+    let(:matching_credit_note) { create(:credit_note, customer:, invoice: create(:invoice, billing_entity:)) }
+    let(:other_credit_note) { create(:credit_note, customer:, invoice: create(:invoice, organization:)) }
+
+    before do
+      matching_credit_note
+      other_credit_note
+    end
+
+    it "returns credit notes with matching billing entity id" do
+      expect(result).to be_success
+      expect(result.credit_notes.pluck(:id)).to contain_exactly matching_credit_note.id
+    end
+  end
```

Comment:
could we haven scenario when we have invoices for more than one billing entity?

Reply:
done 🙌